### PR TITLE
occ core/app command improvement including brute force protection

### PR DESCRIPTION
**Pull request by mmattel**
This PR adds the documentation to configure the Brute Force Protection app via the commandline.

Referencing: doc: https://github.com/owncloud/docs/issues/88 (Brute Force Protection parameters and description needs implementation in occ_app_command)

### CODE REVIEW

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 1]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
 sudo -u www-data ./occ files:scan -p "user_x/files/folder"  
 ....
 
+[[Brute Force Protection-commands]]
+Brute Force Protection Commands
+-------------------------------
+
+Use these commands to configure the Brute Force Protection app.
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: Pls link "the Brute Force Protection app" to the app on the Marketplace.

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 2]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
...
+Parametrisation must be done with the `occ config` command set.
+The combination of `uid` and ÌP address` is used to trigger the ban.
+
+List the current settings
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: Please use [title case](http://www.grammar-monster.com/lessons/capital_letters_title_case.htm) for all headers

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 3]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
...
+sudo -u www-data php occ config:list brute_force_protection
+....
+
+Set the setting
+~~~~~~~~~~~~~~~~
+
+To set a new value, use the command below and replace key and value < > accordingly.
+
+....
+sudo -u www-data php occ config:app:set brute_force_protection <Key> --value=< > --update-only
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: I'd suggest changing `--value=< >` to `--value=<Value>`.

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 4]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+....
+sudo -u www-data php occ config:list brute_force_protection
+....
+
+Set the setting
+~~~~~~~~~~~~~~~~
+
+To set a new value, use the command below and replace key and value < > accordingly.
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: s/and replace key and value < >/and replace `<Key>` and `<Value>`/g

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 5]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
 sudo -u www-data ./occ files:scan -p "user_x/files/folder"  
 ....
 
+[[Brute Force Protection-commands]]
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: s/Brute Force Protection-commands/Brute-Force-Protection-commands/g

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 6]
```diff
@@ -190,6 +190,48 @@ Parameters for short form options will be written either directly after the opti
...
+To set a new value, use the command below and replace key and value < > accordingly.
+
+....
+sudo -u www-data php occ config:app:set brute_force_protection <Key> --value=< > --update-only
+....
+
+Fail Tolerance [attempts]
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Number of wrong attempts to trigger the ban.
+Default:	3
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: This needs to be written as a list or a table for it to render similarly to what the raw text suggests. I suggest a table for the greatest clarity.

**Comment by mmattel** [comment]
@settermjd this is now a bigger change than expected, but cleans really a lot of things that needed to be done. The originals in this repro had errors, some already fixed content was bad again ect. 
There are still some possible fixes needed, but these are more cosmetc things and can be done in a sperarate PR.

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 7]
```diff
@@ -190,21 +192,83 @@ Parameters for short form options will be written either directly after the opti
...
+Brute Force Protection
+----------------------
+
+Marketplace URL: https://marketplace.owncloud.com/apps/brute_force_protection[Brute-Force Protection]
+
+Use these commands to configure the Brute Force Protection app.
+Parametrisation must be done with the `occ config` command set.
+The combination of `uid` and `IP address` is used to trigger the ban.
+
+List the current settings
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: Can you use title case for all headers, pls?

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 8]
```diff
@@ -190,21 +192,83 @@ Parameters for short form options will be written either directly after the opti
...
+....
+sudo -u www-data php occ config:app:set brute_force_protection <Key> --value=<Value> --update-only
+....
+
+Fail Tolerance [attempts]
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Number of wrong attempts to trigger the ban.
+
+[width="80%",cols="30%,70%",]
+|===
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: Nice! I :heart: AsciiDoc tables.

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 9]
```diff
@@ -232,30 +296,24 @@ sudo -u www-data occ s3:create-bucket
 
 Arguments:
 
-[width="70%",cols="30%,70%",]
-|==========================================
-|`bucket` |Name of the bucket to be created
-|==========================================
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: Ah, AsciiDoc, so much simpler than rST. 👍

File: modules/ROOT/pages/configuration/server/occ_app_command.adoc [hunk 10]
```diff
@@ -564,79 +637,48 @@ sudo -u www-data occ user:expire-password
...
-    support@owncloud.com
-....
-
-[NOTE]
-====
-These commands are not available in
-single-user (maintenance) mode <maintenance_commands_label>.
-====
+|===
+|`-a, --all`   | Will add password expiry to all known users.
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_app_command.adoc]: That's a nice touch, formatting the raw text like that. It should make it that much easier to maintain. Thanks.

File: modules/ROOT/pages/configuration/server/occ_command.adoc [hunk 11]
```diff
@@ -228,12 +223,9 @@ If your app has issues, you'll see output like this
 sudo -u www-data php occ app:check-code foo_app
 Analysing /var/www/owncloud/apps/files/foo_app.php
 4 errors
-   line   45: OCP\Response - Static method of deprecated class must not be
-   called
-   line   46: OCP\Response - Static method of deprecated class must not be
-   called
-   line   47: OCP\Response - Static method of deprecated class must not be
-   called
+   line   45: OCP\Response - Static method of deprecated class must not be called
```
Comment (settermjd) [inline modules/ROOT/pages/configuration/server/occ_command.adoc]: Nice cleanup. Thanks!

**Pull request merged**